### PR TITLE
vincent/nicolas review / design decision

### PR DESCRIPTION
**Pull request by vincenthz**

### CODE REVIEW

File: doc/ins_attest_utxo.md [hunk 1]
```diff
@@ -136,10 +138,13 @@ On top of that, Ledger relaxes following checks:
...
 We believe these skipped checks *are not necessary* in order to attest UTxO as the attestation's sole purpose is to **bind** together UTxO (i.e., transaction hash + output number) and amount. This can be done whenever main transaction CBOR structure parses correctly. If the attacker supplies wrong transaction data, either
 1) the Ledger app will show (from the user-perspective) unexpected amounts and the user would not confirm the transaction, or
 2) the Ledger app would show (from the user-perspective) correct amounts but the UTxO has been meddled with. UTxO is, however, referred to by the hash in the transaction under signing and as such, even if Ledger signs such transaction,  *Cardano blockchain nodes* will reject it.
 
+**IOHK vincent/nicolas**: since the transaction size, values of fees, etc. are dynamically updatable, it's probably a bad idea to verify on the ledger side/app. I would consider the addresses as opaque, the validity can be checked on the server side. this also allow a new format more transparently in the future. I don't think it's necessary to add further validity checks like `MAX_ADA`, or `sum(is) > sum(os)`, this will be enforced by the server side anyway.
```
Comment (ppershing) [inline doc/ins_attest_utxo.md]: @vincenthz Perfect, makes sense. Note that we would however enforce check for MAX_ADA during signing (when signing, we need to compute sum of inputs/outputs for computing fee. I would rather avoid `uint64_t` overflows during sum calculation)

File: doc/ins_attest_utxo.md [hunk 2]
```diff
@@ -63,6 +63,8 @@ Upon receiving last txChunk (as determined by the transaction parsing) the respo
 
 Note on HMAC: We use `HMAC_SHA256(key=32 byte session key, message=(txHash,outNum,amount))` to compute the signature digest and return first 16 bytes of it. This should be resilient enough to birthday paradox and other attacks as the key is just per session and we Ledger's signing speed is slow. ❓(IOHK): Is this enough? 
 
+**IOHK vincenthz/nicolas**: we're not quite sure what is this supposed to protect from, and if that's useful for anything until we have the attested data on the ledger validated directly anyway.
```
Comment (ppershing) [inline doc/ins_attest_utxo.md]: @vincenthz @NicolasDP The idea is that attestUtxo will *return* attestation and this attestation is needed in the subsequent `signTx`. The rationale behind this is that Ledger might not be able to store attestation for all UTxOs and so we would stream them one-by-one in signTx call)

Comment (vincenthz) [reply doc/ins_attest_utxo.md]: apologies if this is mentioned somewhere and I've missed it, but I don't understand what's those attestation are for in the first place, nor what they are supposed/trying to protect from.

File: doc/ins_derive_address.md [hunk 3]
```diff
@@ -13,7 +15,7 @@ Note: Unlike BTC Ledger app which returns both public key and the corresponding
 Note: Derive address rejects any path that is not at least an "address" -- currently there is no use for "addresses" representing whole account (e.g., path 44'/1815'/0') nor addresses representing chain (e.g., path 44'1815'/0'/0)
 
 ❓(IOHK): Should we restrict this to only to valid chain number (e.g., internal/external)? Or even just to an external chain? (Note: There probably isn't much reason to let the user verify change addresses but we recommend allowing it anyway)
-
+**IOHK nicolas/vincent**: you might want to provide the application a way to recover the change address (this will be useful for wallet recovery). So you need to restrict it to internal/external addresses but still allow the internal address too.
```
[the remark this answers is not in the document]
Comment (vincenthz) [reply doc/ins_derive_address.md]: good point. I think just having this as debugging capabilities is still useful, but clearly we would directly derive from xprv on the host.

**Pull request merged**